### PR TITLE
feat: first iteration of C client library

### PR DESCRIPTION
**Pull request by letFunny**

### CODE REVIEW

File: src/client.c [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+int dqlite_finalize(dqlite_stmt *stmt)
+{
+	struct client_context context;
+
+	if (stmt == NULL) {
+		return SQLITE_OK;
+	}
+	// TODO add options.
+	clientContextMillis(&context, 5000);
+	if (clientSendFinalize(&stmt->proto, stmt->stmt_id, &context) !=
```
[the remark this answers is not in the document]
Comment (cole-miller) [reply src/client.c]: I don't feel strongly about it personally but I would like to stick to it in new code for consistency.

[the remark this answers is not in the document]
Comment (letFunny) [reply src/client.c]: makes sense, changed!

File: test/integration/test_client_protocol.c [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+ * Handle client requests
+ *
+ ******************************************************************************/
+
+SUITE(client_protocol);
+
+static char *bools[] = {"0", "1", NULL};
+
+static MunitParameterEnum client_params[] = {
+    {"disk_mode", bools},
```
[the remark this answers is not in the document]
Comment (letFunny) [reply test/integration/test_client_protocol.c:18]: This file is copied from existing tests, just renamed to better reflect what it is still testing. Should I remove the param before the old disk mode goes away?

**Comment by letFunny** [comment]
This is not a priority for the time being.

**Pull request closed without merging**